pkg/dogstatsd/server_test.go

```diff
@@ -572,27 +577,39 @@ func TestDebugStatsSpike(t *testing.T) {
 	}
 
 	send(10)
-	time.Sleep(1050 * time.Millisecond)
+
+	clk.Add(1050 * time.Millisecond)
 	send(10)
-	time.Sleep(1050 * time.Millisecond)
+
+	clk.Add(1050 * time.Millisecond)
 	send(10)
-	time.Sleep(1050 * time.Millisecond)
+
+	clk.Add(1050 * time.Millisecond)
 	send(10)
-	time.Sleep(1050 * time.Millisecond)
+
+	clk.Add(1050 * time.Millisecond)
 	send(500)
 
 	// stop the debug loop to avoid data race
 	s.DisableMetricsStats()
-	time.Sleep(500 * time.Millisecond)
+	time.Sleep(1 * time.Millisecond)
+
 	assert.True(s.hasSpike())
 
 	s.EnableMetricsStats()
-	time.Sleep(1050 * time.Millisecond)
+	// This sleep is necessary as we need to wait for the goroutine function wihtin 'EnableMetricsStats' to start.
+	// If we remove the sleep, the debug loop ticker will not be triggered by the clk.Add() call and the 500 samples
+	// added with 'send(500)' will be considered as if they had been added in the same second as the previous 500 samples.
+	// This will lead to a spike because we have 1000 samples in 1 second instead of having 500 and 500 in 2 different seconds.
```

Comment:
This is pretty gross, but yeah, it's the best we can do in Go :(

Reply:
Another option is to use [`assert.Eventually`](https://pkg.go.dev/github.com/stretchr/testify/assert#Eventually) which just really hides the sleep call.

Reply:
In this case I don't think `assert.Eventually` would help because the `time.Sleep` is needed before the `clk.Add()` call.